lib/datadog/core/configuration/option_definition.rb

```diff
@@ -11,9 +11,9 @@ class OptionDefinition
 
         attr_reader \
           :default,
+          :experimental_default_proc,
```

Comment:
Here or somewhere on this file, I think it would be useful to document (or repeat from the PR description) the reason why we're calling this experimental, and what are our thoughts concerning the next steps (e.g. either removing it, or renaming it back to non-experimental).